### PR TITLE
Actually specify the correct CUDA architecture using SYCL

### PR DESCRIPTION
**Pull request by masterleinad**
Even if we can build without it currently, it just seems to be right to make sure we are building for the correct architecture.

### CODE REVIEW

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by dalg24** [comment]
SYCL build does not pass

**Comment by masterleinad** [comment]
> SYCL build does not pass

Yes, that doesn't make any sense to me.

**Comment by masterleinad** [comment]
The only failing test gives
```
[ RUN      ] sycl.Random_XorShift64
Test Seed:1618432068767902210
Test Scalar=int
 -- Testing randomness properties
Pass: 1 1 -0.000104996 -0.000125552 3.28249e-05 || 0.000139188
 -- Testing 1-D histogram
Density 1D: 6.43252e-07 -0.784139 -0.0910377 || 0.051031 27299 30320 || 132837 28674.4 || -1258.51 -2.0744
 -- Testing 3-D histogram
Density 3D: 6.43252e-07 -0.78388 -0.00210329 || 0.051031 1e+64 -1e+64
/var/jenkins/workspace/Kokkos/algorithms/unit_tests/TestRandom.hpp:411: Failure
Value of: 1
Expected: test_int.pass_hist1d_var
Which is: 0
Test Seed:1618432068922778626
Test Scalar=int
 -- Testing randomness properties
Pass: 1 1 -4.31214e-05 0.000134111 -1.42102e-05 || 0.000139188
 -- Testing 1-D histogram
Density 1D: 6.43252e-07 -0.778921 0.129904 || 0.051031 27243 30329 || 129702 28674.4 || 1795.79 -2.0744
 -- Testing 3-D histogram
Density 3D: 6.43252e-07 -0.783137 -0.034169 || 0.051031 1e+64 -1e+64
/var/jenkins/workspace/Kokkos/algorithms/unit_tests/TestRandom.hpp:411: Failure
Value of: 1
Expected: test_int.pass_hist1d_var
Which is: 0
[  FAILED  ] sycl.Random_XorShift64 (316 ms)
```

**Comment by dalg24** [comment]
Are you still working on this?

**Comment by masterleinad** [comment]
I would like to figure out why this would make any difference but I'm currently clueless and it's not a high priority.

**Comment by dalg24** [comment]
Note that https://github.com/kokkos/kokkos/pull/4117#issue-675608870 provided a nice rational for this patch

**Comment by masterleinad** [comment]
From https://github.com/kokkos/kokkos/pull/4117#issue-675608870:
> When compiling with the DPC++ CUDA backend all kernels are compiled in advance. However if a kernel is launched on a device it was not built with the matching arch for, it will be recompiled at runtime.
> Passing the --cuda-gpu-arch flag as a sycl-backend-target flag will add it to the cuda program compile options, so it's initially built for the correct arch and doesn't need recompilation at runtime.
> [...]

**Comment by joeatodd** [comment]
Hey, just wanted to add a brief note here to say that I've been building using this patch & it works at our end. I plan to take a look at this CI test failure at some point & see if I can figure out why it might fail. Will update if/when I get round to that.

**Comment by masterleinad** [comment]
With https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/FreeFunctionQueries/SYCL_INTEL_free_function_queries.asciidoc, we are now able to mirror the HIP/CUDA implementation.

**Comment by masterleinad** [comment]
Retest this please.

File: algorithms/unit_tests/TestRandom.hpp [hunk 1]
```diff
@@ -327,10 +327,6 @@ template <class RandomGenerator, class Scalar>
 struct test_random_scalar {
   using rnd_type = typename RandomGenerator::generator_type;
 
-  int pass_mean, pass_var, pass_covar;
-  int pass_hist1d_mean, pass_hist1d_var, pass_hist1d_covar;
-  int pass_hist3d_mean, pass_hist3d_var, pass_hist3d_covar;
-
```
Comment (dalg24) [inline algorithms/unit_tests/TestRandom.hpp:333]: These changes look good but why did you push them here?

**Review by dalg24 — APPROVED** [review]

File: algorithms/unit_tests/TestRandom.hpp (hunk 1 above)
Comment (masterleinad) [reply algorithms/unit_tests/TestRandom.hpp:333]: I was trying to debug the issue and wanted better error messages. I thought it was reasonable to keep these changes but I can move them somewhere else if you prefer.

Comment (dalg24) [reply algorithms/unit_tests/TestRandom.hpp:333]: It's ok by me.

**Review by nliber — APPROVED** [review]

**Pull request merged**